File: 프로그래머스/1/42840. 모의고사/모의고사.java
class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int size = answers.length;
        
        
        int[] one = { 1, 2, 3, 4, 5};
        int[] two = { 2, 1, 2, 3, 2, 4, 2, 5};
        int[] three = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        int[] score = {0,0,0};
        
        for(int i = 0; i < size; i++){
            if(one[i%5] == answers[i])
                score[0]+=1;
            if(two[i%8] == answers[i])
                score[1]+=1;
            if(three[i%10] == answers[i])
                score[2]+=1;
        }
        
        int maxScore = Math.max(score[0], Math.max(score[1], score[2]));
        
        int count = 0;
        for (int i = 0; i < 3; i++) {
            if (score[i] == maxScore) {
                count++;
            }
        }
        
        int[] result = new int[count];
        int index = 0;
        for (int i = 0; i < 3; i++) {
            if (score[i] == maxScore) {
                result[index++] = i + 1;
            }
        }
        
        return result;
    }
}

File: JavaScript/프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    var answer = [];
    const count = [0, 0, 0]
    const pattern = [
        [1,2,3,4,5], 
        [2,1,2,3,2,4,2,5], 
        [3,3,1,1,2,2,4,4,5,5]]
    
    for(let i = 0; i < answers.length; i++){
        if(pattern[0][i%pattern[0].length] === answers[i]){
            count[0] += 1
        }
        if(pattern[1][i%pattern[1].length] === answers[i]){
            count[1] += 1
        }
        if(pattern[2][i%pattern[2].length] === answers[i]){
            count[2] += 1
        }
    }
    
    const max = Math.max(...count)
    for(let i = 0; i < 3; i++){
        if(max === count[i]) answer.push(i+1)
    }
    return answer;
}